PWGHF/vertexingHF/AliAnaTaskSEXic0SL.C

```diff
@@ -0,0 +1,947 @@
+		const bool t_trigFiredMB   = (fEvtTrig & AliVEvent::kINT7)?true:false;
+		const bool t_trigFiredHMV0 = (fEvtTrig & AliVEvent::kHighMultV0)?true:false;
+
+		fHisto->FillTH1("EvtTrig", "Any", 1); //@
+		if (t_trigFiredMB) fHisto->FillTH1("EvtTrig", "MB", 1); //@
+		if (t_trigFiredHMV0) fHisto->FillTH1("EvtTrig", "HMV0", 1); //@
+
+		//Activate RDHF cut by invoking it, then check pileup status by using multi vertexer
+		const bool t_evtGoodMB   = fEvtCutsMB  ->IsEventSelected(fEvt);
+		const bool t_evtGoodHMV0 = fEvtCutsHMV0->IsEventSelected(fEvt);
+		fEvtPileupMB   = fEvtCutsMB  ->IsEventRejectedDueToPileup();
+		fEvtPileupHMV0 = fEvtCutsHMV0->IsEventRejectedDueToPileup();
+		if (fEvtPileupMB && fEvtPileupHMV0) return; //Return 8d
+		else fHisto->FillTH1("EvtCut", "PileupX>0", 1); //@
+
+		//Check vtx validity (NOT vtxZ)
+		fEvtVtx = fEvt->GetPrimaryVertex();
+		if (!fEvtVtx || fEvtVtx->GetNContributors()<cut_vtxNContributors) return; //Return 9d
+
+		//Store AliNormalizationCounter: DO NOT CHANGE THIS POSITION! (after vtx found, before |VtxZ| < 10)
+		if (t_trigFiredMB)
+		{
+			if (fEvtMult >= 0.0 && fEvtMult <= 100.0)
+			{
+				fANC_MB_0to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+				if (fEvtINELLgt0) fANC_INEL0_MB_0to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+			}
+			if (fEvtMult >= 30.0 && fEvtMult <= 100.0)
+			{
+				fANC_MB_30to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+				if (fEvtINELLgt0) fANC_INEL0_MB_30to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+			}
+			if (fEvtMult >= 0.1 && fEvtMult <= 30.0)
+			{
+				fANC_MB_0p1to30->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+				if (fEvtINELLgt0) fANC_INEL0_MB_0p1to30->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+			}
+		}
+		if (t_trigFiredHMV0 && fEvtMult >= 0.0 && fEvtMult <= 0.1)
+		{
+			fANC_HMV0_0to0p1->StoreEvent(fEvt, fEvtCutsHMV0, IsMC);
+			if (fEvtINELLgt0) fANC_INEL0_HMV0_0to0p1->StoreEvent(fEvt, fEvtCutsHMV0, IsMC);
+		}
+
+		//Check vtxZ
+		fEvtVtxZ = fEvtVtx->GetZ();
+		if (fabs(fEvtVtxZ) > cut_vtxZ) return; //Return 10d
+		else fHisto->FillTH1("EvtCut", "VtxZ10", 1); //@
+
+		//Check INEL>0
+		if (AliPPVsMultUtils::IsINELgtZERO(fEvt)==true)	fEvtINELLgt0 = true;
+		if (fEvtINELLgt0) fHisto->FillTH1("EvtCut", "INEL>0", 1); //@
+	}//data
+	else //MC
+	{
+		fMCEvt = fInputHandler->MCEvent();
+
+		//Check vtxZ
+		fEvtVtx = fMCEvt->GetPrimaryVertex();
+		fEvtVtxZ = fEvtVtx->GetZ();
+		if (fabs(fEvtVtxZ) > cut_vtxZ) return; //Return 7mc
+
+		//Get true Xic0
+		int nXic0 = 0;
+		const int nTracksMC = fMCEvt->GetNumberOfTracks();
+		for (int a=0; a<nTracksMC; a++)
+		{
+			fMCPart = (AliAODMCParticle*)fMCEvt->GetTrack(a);
+			if (!fMCPart || abs(fMCPart->GetPdgCode())!=PDGCode_Xic0) continue; //Continue 1
+
+			//Search e-Xi pair among daughters
+			bool t_found_e = false;
+			bool t_found_Xi = false;
+			AliAODMCParticle* fMCPart_e = 0;
+			AliAODMCParticle* fMCPart_Xi = 0;
+			for (int b=fMCPart->GetDaughterFirst(); b<=fMCPart->GetDaughterLast(); b++)
```

Comment:
You need to check if the numbers of daughters is the number you want. Right now you also select the four-body decays

Reply:
1. I checked by comparing "manual counting over loop <-> AliAODMCParticle->GetNDaughters()".
They're always the same. I'll update it to use GetNDaughters(), as it's more simple.

2. I added a boolean flag if either pi0 or charged pi is found among daughters.

Reply:
(I don't fully follow your 1) and 2), please correct me if I understood wrongly)

1) It might be the case that for the MC you currently analysing all Xic have N=3 daughters, since the four-body Xic decays were not yet enabled when producing these MCs. However, as we are discussing via email, you need to request a new MC with additional Xic decays enabled for your analysis to do the background-subtraction template fit. And for that MC this will not hold anymore. So please add a condition on the correct number of Xic daughters when you tag Xic for the efficiency calculation.

2) Ok for calculating the reconstruction efficiency. I don't think however that labels of the truth Xic0 and e/Xi's mothers are enough to perform the background template fit. You also need to tag which four-body decay channel it is.

Reply:
Ok, now I got your point. I updated the truth MC handling as follows.

1. Possible 4-body decay modes are these two:
 a. Xic+ -> (e+) + (nu) + (Xi*0) -> (e+) + (nu) + (Xi- + pi+)
 b. Xic0 -> (e-) + (nu_bar) + (Xi*+) -> (e-) + (nu_bar) + (Xi+ + pi0)

2. At the truth level,
 a. Truth Xic+ also will be accepted in addition to the Xic0
 b. Added label and PDG of truth particle
 
The distinction should be possible If I compare 2-b with reco. electron's (or Xi) mother label/PDG.
For the case of 4-body decay only, requiring Xi's mother PDG to be either Xi*0 or Xi*+ should be sufficient.